bundles/automation/org.eclipse.smarthome.automation.module.script/src/main/java/org/eclipse/smarthome/automation/module/script/internal/handler/AbstractScriptModuleHandler.java

```diff
@@ -106,10 +107,13 @@ private void loadConfig() {
     protected void setExecutionContext(ScriptEngine engine, Map<String, ?> context) {
         ScriptContext executionContext = engine.getContext();
 
-        // make the whole context available as "ctx"
+        // Add the rule's UID to the context
+        final Map<String, Object> contextNew = new HashMap<>(context);
+        contextNew.put("ruleUID", this.ruleUID);
+
+        // make the whole (new) context available as "ctx"
         // Note: We don't use "context" here as it doesn't work on all JVM versions!
-        executionContext.setAttribute("ctx", context, ScriptContext.ENGINE_SCOPE);
-        executionContext.setAttribute("ruleUID", this.ruleUID, ScriptContext.ENGINE_SCOPE);
```

Comment:
@triller-telekom You are right, I moved ruleUID from global namespace to ctx (so removed it from the global namespace).
#3007 contains the discussion to add ruleUID to "ctx".
#1783 moved ruleUID from "ctx" to the global namespace without any discussion or agreement (IMHO).
I assume (reading your comment) you are using "ruleUID" from the global now as I use it from "ctx".

You are right, the comment of @smerschjohann indicates that it should be also in the global namespace.

So, what should we do?